pyensign/exceptions.py

```diff
@@ -17,3 +17,27 @@ class AuthenticationError(PyEnsignError):
     """
 
     pass
+
+
+class EnsignError(PyEnsignError):
+    """
+    Raised when PyEnsign receives an error from the Ensign server
+    """
+
+    pass
```

Comment:
It might be nice to have the status code and the message stored on the error (and authentication error) just to make testing easier in the future. But this can be a story we create to do later.